docs/src/main/asciidoc/lra-guide.adoc

```diff
@@ -0,0 +1,196 @@
+(https://github.com/eclipse/microprofile-lra/blob/master/api/src/main/java/org/eclipse/microprofile/lra/annotation/ws/rs/LRA.java[`@LRA`]).
+When such a method is called a context is created (if one is not already present) which is passed
+along with subsequent JAX-RS invocations until a method is reached
+which also contains an `@LRA` annotation with an attribute that indicates that the LRA should be
+closed or cancelled. The default is for the LRA to be closed in the same method that started the
+LRA (which itself may have propagated the context during method execution).
+The JAX-RS resource indicates that it wishes to participate in the interaction by, minimally,
+marking one of the methods with an
+https://github.com/eclipse/microprofile-lra/blob/master/api/src/main/java/org/eclipse/microprofile/lra/annotation/Compensate.java[`@Compensate`]
+annotation. If the context is later cancelled then this compensate action is guaranteed to be
+called even in the presence of failures and is the trigger for the resource to compensate for any
+activities it performed in the context of the LRA. This guarantee enables services to operate
+reliably with the assurance of eventual consistency (when all compensation activities have
+ran to completion).
+
+Note that this version of the extension only supports JAX-RS based services (and a subsequent
+revision will include support for non JAX-RS based services).
+
+== Configuration
+
+Once you have your Quarkus maven project configured you can add the `narayana-lra` extension
+by running the following command in your project base directory:
+
+[source,bash]
+----
+./mvnw quarkus:add-extension -Dextensions="narayana-lra"
+----
+
+This will add the following to your pom.xml
+
+[source,xml]
+----
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-narayana-lra</artifactId>
+    </dependency>
+----
+
+If there is a running coordinator then this is all you need in order to create
+new LRAs and to enlist participants with them.
+
+The LRA extension can be configured by placing an `application.properties` file
+in the `src/main/resources` directory. The only LRA specific property is
+`quarkus.lra.coordinator-url=<url>` which specifies the HTTP endpoint of an external
+coordinator, for example:
+
+[source,bash]
+----
+quarkus.lra.coordinator-url=http://localhost:8080/lra-coordinator
+----
+
+For a narayana coordinator the path component of the url is normally `lra-coordinator`.
+Coordinators can be obtained from `https://hub.docker.com/r/jbosstm/lra-coordinator`
+or you can build your own coordinator using a maven pom that includes the appropriate
+dependencies. A quarkus quickstart will be provided to show how to do this or you can
+take a look at one of the https://github.com/jbosstm/quickstart/tree/master/rts/lra-examples/lra-coordinator[narayana quickstarts].
+Another option would be to run it managed inside a WildFly application server.
```

Comment:
Oh... that lead to dev services :-D 

We may want to automatically start a coordinator in dev/test modes.

Reply:
How do I do that. In the integration test I use @QuarkusTestResource to hook into the test lifecycle and I start a coordinator in a docker container?

Reply:
Dev Services is kind of the same thing but run during test and dev mode. We can do that in another PR (a few things are still in flux around dev services). Can you open an issue so we don't forget?

Reply:
#18425